diff --git a/browser/branding/unofficial/configure.sh b/browser/branding/unofficial/configure.sh
index edd3bd3e86e50..e197b3e08d51b 100644
--- a/browser/branding/unofficial/configure.sh
+++ b/browser/branding/unofficial/configure.sh
@@ -3,3 +3,8 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 MOZ_APP_DISPLAYNAME=Nightly
+
+##############################################################################
+# ssb: modified
+##############################################################################
+MOZ_APP_UA_NAME=Firefox
diff --git a/browser/branding/unofficial/pref/firefox-branding.js b/browser/branding/unofficial/pref/firefox-branding.js
index f22dd0a51ccb3..243e8d54c9168 100644
--- a/browser/branding/unofficial/pref/firefox-branding.js
+++ b/browser/branding/unofficial/pref/firefox-branding.js
@@ -30,3 +30,43 @@ pref("app.update.badgeWaitTime", 0);
 // Number of usages of the web console.
 // If this is less than 5, then pasting code into the web console is disabled
 pref("devtools.selfxss.count", 5);
+
+
+/**
+ * ssb: modified
+ */
+
+// Required to work encrypt DB
+pref("security.nocertdb", false);
+// pref("general.useragent.compatMode.firefox", true, locked); // Do on configure. See browser/branding/unofficial/configure.sh
+
+// Updater
+pref("app.update.auto", false);
+
+// privacy
+pref("breakpad.reportURL", "", locked);
+pref("browser.tabs.crashReporting.sendReport", false, locked);
+pref("browser.crashReports.unsubmittedCheck.enabled",	false, locked);
+pref("toolkit.telemetry.archive.enabled", false, locked);
+pref("toolkit.telemetry.bhrPing.enabled", false, locked);
+pref("toolkit.telemetry.enabled", false, locked);
+pref("toolkit.telemetry.firstShutdownPing.enabled", false, locked);
+pref("toolkit.telemetry.geckoview.streaming", false, locked);
+pref("toolkit.telemetry.newProfilePing.enabled", false, locked);
+pref("toolkit.telemetry.pioneer-new-studies-available", false, locked);
+pref("toolkit.telemetry.reportingpolicy.firstRun", false, locked);
+pref("toolkit.telemetry.server", "", locked);
+pref("toolkit.telemetry.shutdownPingSender.enabled", false, locked);
+pref("toolkit.telemetry.shutdownPingSender.enabledFirstSession", false, locked);
+pref("toolkit.telemetry.testing.overrideProductsCheck", false, locked);
+pref("toolkit.telemetry.unified", false, locked);
+pref("toolkit.telemetry.updatePing.enabled", false, locked);
+pref("dom.private-attribution.submission.enabled", false, locked);
+pref("app.shield.optoutstudies.enabled", false, locked);
+pref("browser.crashReports.unsubmittedCheck.autoSubmit2", false, locked);
+pref("privacy.trackingprotection.origin_telemetry.enabled", false, locked);
+pref("privacy.userContext.enabled", true);
+pref("privacy.userContext.ui.enabled", true);
+pref("trailhead.firstrun.branches", "", locked);
+pref("extensions.webcompat-reporter.enabled", false, locked);
+pref("browser.discovery.enabled", false);
