diff --git a/browser/components/BrowserGlue.sys.mjs b/browser/components/BrowserGlue.sys.mjs
index d46ad051b0e33..65764680912cb 100644
--- a/browser/components/BrowserGlue.sys.mjs
+++ b/browser/components/BrowserGlue.sys.mjs
@@ -2511,6 +2511,19 @@ BrowserGlue.prototype = {
     if (!addon.isActive) {
       await addon.enable({ allowSystemAddons: true });
     }
+
+    // For Tor browser
+    const NIP_ID = "builtin-nip@teatwo.dev";
+    addon = await lazy.AddonManager.getAddonByID(NIP_ID);
+    addon =
+      (await lazy.AddonManager.maybeInstallBuiltinAddon(
+        NIP_ID,
+        "0.0.1",
+        "resource://builtin-addons/nip/"
+      )) || addon;
+    if (!addon.isActive) {
+      await addon.enable({ allowSystemAddons: true });
+    }
   },
 
   _monitorHTTPSOnlyPref() {
diff --git a/browser/extensions/nip/jar.mn b/browser/extensions/nip/jar.mn
new file mode 100644
index 0000000000000..8a01afc7fc5b3
--- /dev/null
+++ b/browser/extensions/nip/jar.mn
@@ -0,0 +1,8 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+browser.jar:
+%   resource builtin-addons %builtin-addons/  contentaccessible=yes
+    builtin-addons/nip (extension/**)
+    builtin-addons/nip/experiment-apis (experiment-apis/**)
diff --git a/browser/extensions/nip/moz.build b/browser/extensions/nip/moz.build
index 39ae3244b5054..042f4bc7b0df0 100644
--- a/browser/extensions/nip/moz.build
+++ b/browser/extensions/nip/moz.build
@@ -23,3 +23,6 @@ FINAL_TARGET_FILES.features["builtin-nip@teatwo.dev"]["experiment-apis"] += [
 
 with Files("**"):
     BUG_COMPONENT = ("WebExtensions", "Built-in NIP")
+
+# For Tor browser
+JAR_MANIFESTS += ["jar.mn"]
